compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -446,10 +448,77 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                 }
             }
         }
+
         err.span_label(decl_span, "binding declared here but left uninitialized");
+        if show_assign_sugg {
+            struct LetVisitor {
+                decl_span: Span,
+                ty_span: Option<Span>,
+            }
+
+            impl<'v> Visitor<'v> for LetVisitor {
+                fn visit_stmt(&mut self, ex: &'v hir::Stmt<'v>) {
+                    if self.ty_span.is_some() {
+                        return;
+                    }
+                    if let hir::StmtKind::Local(hir::Local {
+                            span, init: None, ty: Some(ty), ..
+                        }) = &ex.kind && span.contains(self.decl_span) {
+                            self.ty_span = Some(ty.span);
+                    }
+                    hir::intravisit::walk_stmt(self, ex);
+                }
+            }
+
+            let mut visitor = LetVisitor { decl_span, ty_span: None };
+            visitor.visit_body(&body);
+            if let Some(ty_span) = visitor.ty_span {
+                self.suggest_assign_value(&mut err, moved_place, ty_span);
+            }
+        }
         err
     }
 
+    fn suggest_assign_value(
+        &self,
+        err: &mut Diagnostic,
+        moved_place: PlaceRef<'tcx>,
+        ty_span: Span,
+    ) {
+        let ty = moved_place.ty(self.body, self.infcx.tcx).ty;
+        debug!("ty: {:?}, kind: {:?}", ty, ty.kind());
+
+        let assign_value = match ty.kind() {
```

Comment:
None of the strings are being formatted anymore, all branches could be changed to return a &'static str.